compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaServiceDeclarationSymbol.java

```diff
@@ -216,4 +218,42 @@ public BallerinaServiceDeclarationSymbol build() {
                                                          this.qualifiers, this.annots, this.bSymbol, this.context);
         }
     }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.getInternalSymbol().getName().getValue(),
+                this.getLocation().map(Location::lineRange).orElse(null));
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (!(obj instanceof ServiceDeclarationSymbol)) {
+            return false;
+        }
+
+        BallerinaServiceDeclarationSymbol symbol = (BallerinaServiceDeclarationSymbol) obj;
+        return nameEquals(symbol.getInternalSymbol().getName().getValue())
+                && isSameLocation(this.getLocation(), symbol.getLocation());
```

Comment:
Checking `isSameLocation()` is not enough. We have to check whether the module is the same too since the location is given relative to the module directory. Hence, there can be instances where locations are equal even though the 2 symbols are from different modules.

Also, let's change the visibility of the `isSameLocation()` and `isSameModule()` to `protected` and reuse them.